libs/server/Resp/Parser/RespCommand.cs

```diff
@@ -343,6 +350,17 @@ public static ulong OneIfRead(this RespCommand cmd)
             return Unsafe.As<bool, byte>(ref inRange);
         }
 
+        /// <summary>
+        /// Returns True if <paramref name="cmd"/> does not change state of database
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsNonDbStateMutatingCommand(this RespCommand cmd, CustomCommand customCommand, CustomObjectCommand customObjectCommand)
+        {
+            return (cmd == RespCommand.CustomCmd && customCommand?.type == CommandType.Read) || (cmd == RespCommand.CustomObjCmd && customObjectCommand?.type == CommandType.Read) ||
+                cmd <= RespCommand.ZSCORE || // command is from read section
```

Comment:
This can lead to a case where we return a value in the get command but the set for that key has not persisted. So if the server crashes before persisting it that key will become unavailable.

Reply:
Good point, query-like commands won't be lost if the "enqueued" AOF items that need to be flushed are not related to the key the queries are related to.
However, for the same key, not waiting for persistence could return a value for a Key-Val that is failed to persist since it will be in memory.

I guess I could look into having some sort of way to know if something is a dirty/new (not flushed) that is being queried and then proceed off of that.

Reply:
Dirty Read handling will be done in separate PR, limited the set of Commands in this first PR